tests/bluetooth/iso/modules/central.c

```diff
@@ -0,0 +1,476 @@
+{
+	int ret;
+	uint8_t channel_index;
+
+	ret = channel_index_get(conn, &channel_index);
+	if (ret) {
+		LOG_ERR("Channel index not found");
+	}
+
+	ret = bt_gatt_dm_start(conn, BT_UUID_NUS_SERVICE, &discovery_cb,
+			       &nus_client[channel_index]);
+	return ret;
+}
+
+static uint8_t ble_data_received(struct bt_nus_client *nus, const uint8_t *data, uint16_t len)
+{
+	LOG_HEXDUMP_INF(data, len, "NUS received:");
+	return BT_GATT_ITER_CONTINUE;
+}
+
+static int nus_client_init(void)
+{
+	int ret;
+	struct bt_nus_client_init_param init = {.cb = {
+							.received = ble_data_received,
+						}};
+
+	for (int i = 0; i < ARRAY_SIZE(nus_client); i++) {
+		ret = bt_nus_client_init(&nus_client[i], &init);
+		if (ret) {
+			LOG_ERR("NUS Client [%d] initialization failed, ret = %d", i, ret);
+			return ret;
+		}
+	}
+
+	return 0;
+}
+
+static int device_found(uint8_t type, const uint8_t *data, uint8_t data_len,
+			const bt_addr_le_t *addr)
+{
+	int ret;
+	struct bt_conn *conn;
+
+	if (all_peers_connected()) {
+		LOG_DBG("All headset connected");
+		return 0;
+	}
+
+	if ((data_len == REMOTE_DEVICE_NAME_PEER_LEN) &&
+	    (strncmp(REMOTE_DEVICE_NAME_PEER, data, REMOTE_DEVICE_NAME_PEER_LEN) == 0)) {
+		LOG_DBG("Device found");
+
+		ret = bt_le_scan_stop();
+		if (ret && ret != -EALREADY) {
+			LOG_WRN("Stop scan failed: %d", ret);
+		}
+
+		ret = bt_conn_le_create(addr, BT_CONN_LE_CREATE_CONN, &conn_param, &conn);
+		if (ret) {
+			LOG_ERR("Could not init connection");
+			start_scan();
+			return ret;
+		}
+
+		for (int i = 0; i < CONFIG_BT_MAX_CONN; i++) {
+			if (remote_peer[i].conn == NULL) {
+				remote_peer[i].conn = conn;
+				break;
+			}
+		}
+
+		return 0;
+	}
+
+	return -ENOENT;
+}
+
+/** @brief  Parse BLE advertisement package.
+ */
+static void ad_parse(struct net_buf_simple *p_ad, const bt_addr_le_t *addr)
+{
+	while (p_ad->len > 1) {
+		uint8_t len = net_buf_simple_pull_u8(p_ad);
+		uint8_t type;
+
+		/* Check for early termination */
+		if (len == 0) {
+			return;
+		}
+
+		if (len > p_ad->len) {
+			LOG_WRN("AD malformed");
+			return;
+		}
+
+		type = net_buf_simple_pull_u8(p_ad);
+
+		if (device_found(type, p_ad->data, len - 1, addr) == 0) {
+			return;
+		}
+
+		(void)net_buf_simple_pull(p_ad, len - 1);
+	}
+}
+
+static void on_device_found(const bt_addr_le_t *addr, int8_t rssi, uint8_t type,
+			    struct net_buf_simple *p_ad)
+{
+	if ((type == BT_GAP_ADV_TYPE_ADV_IND || type == BT_GAP_ADV_TYPE_EXT_ADV)) {
+		/* Note: May lead to connection creation */
+		ad_parse(p_ad, addr);
+	}
+}
+
+static void start_scan(void)
+{
+	int ret;
+
+	ret = bt_le_scan_start(BT_LE_SCAN_PASSIVE, on_device_found);
+	if (ret && ret != -EALREADY) {
+		LOG_WRN("Scanning failed to start: %d", ret);
+		return;
+	}
+
+	LOG_INF("Scanning successfully started");
+}
+
+static bool all_peers_connected(void)
+{
+	for (int i = 0; i < ARRAY_SIZE(remote_peer); i++) {
+		if (remote_peer[i].conn == NULL) {
+			return false;
+		}
+	}
+
+	return true;
+}
```

Comment:
This can be moved to the top of the file, that way you don't need to declare it on line 42